manifests/agent.pp

```diff
@@ -51,21 +51,19 @@
 # "somehost-1".
 #
 define bamboo_agent::agent(
-  $id           = $title,
-  $home         = "${bamboo_agent::install_dir}/agent${title}-home",
-  $wrapper_conf_properties = {},
-  $manage_capabilities     = false,
-  $capabilities            = {},
-  $expand_id_macros        = true,
-  $private_tmp_dir         = false,
-  $refresh_service         = false,
-  $user_name               = $bamboo_agent::user_name,
-  $group                   = $bamboo_agent::user_group,
+  $id                           = $title,
+  $home                         = "${bamboo_agent::install_dir}/agent${title}-home",
+  Hash $wrapper_conf_properties = {},
+  Boolean $manage_capabilities  = false,
+  Boolean $manage_service       = true,
+  Hash $capabilities            = {},
+  Boolean $expand_id_macros     = true,
+  Boolean $private_tmp_dir      = false,
+  Boolean $refresh_service      = false,
+  $user_name                    = $bamboo_agent::user_name,
+  $group                        = $bamboo_agent::user_group,
 ){
 
-  validate_hash($wrapper_conf_properties)
```

Comment:
I wonder why these lines are gone

Reply:
those functions are obsolete and generate warnings in catalog
Equivalent is setting attribute type accordingly